practice_06.15.2019_generics/Task3_LinkedList/LinkedList.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using practice_06._15._2019_generics.Task1_MyList;
+
+namespace practice_06._15._2019_generics.Task3_LinkedList
+{
+    public class LinkedList<T> : IEnumerable<T>
+    {
+        Node<T> head;
+        Node<T> tail;
+        int count;
+
+        public void Add(T data)
+        {
+            Node<T> node = new Node<T>(data);
+
+            if (head == null)
+                head = node;
+            else
+                tail.Next = node;
+
+            tail = node;
+            count++;
+        }
+
+        public bool Remove(T data)
+        {
+            Node<T> current = head;
+            Node<T> previous = null;
+
+            while (current != null)
+            {
+                if (current.Data.Equals(data))
+                {
+                    if (previous != null)
+                    {
+                        previous.Next = current.Next;
+
+                        if (current.Next == null)
+                            tail = previous;
+                    }
+                    else
+                    {
+                        head = head.Next;
+
+                        if (head == null)
+                            tail = null;
+                    }
+                    count--;
+                    return true;
+                }
+
+                previous = current;
+                current = current.Next;
+            }
+            return false;
+        }
+
+        public int Count { get { return count; } }
+        public bool IsEmpty { get { return count == 0; } }
+
+        public void Clear()
+        {
+            head = null;
+            tail = null;
+            count = 0;
+        }
+
+        public bool Contains(T data)
+        {
+            Node<T> current = head;
+            while (current != null)
+            {
+                if (current.Data.Equals(data))
+                    return true;
+                current = current.Next;
+            }
+            return false;
+        }
+
+        public void AppendFirst(T data)
+        {
+            Node<T> node = new Node<T>(data);
+            node.Next = head;
+            head = node;
+            if (count == 0)
+                tail = head;
+            count++;
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return ((IEnumerable)this).GetEnumerator();
+        }
+
+        IEnumerator<T> IEnumerable<T>.GetEnumerator()
+        {
+            Node<T> current = head;
+            while (current != null)
+            {
+                yield return current.Data;
+                current = current.Next;
+            }
+        }
+    }
+
+    public class Node<T>
+    {
+        public Node(T data)
+        {
+            Data = data;
+        }
+        public T Data { get; set; }
+        public Node<T> Next { get; set; }
```

Comment:
All members should be separated with an empty line.

Reply:
fixed